records/generator.go

```diff
@@ -81,8 +81,14 @@ func (rg *RecordGenerator) ParseState(leader string, c Config) error {
 		return err
 	}
 
+	hostSpec := labels.HostNameSpec1123
+	if c.EnforceDNS952 {
+		hostSpec = labels.HostNameSpec952
+	}
+	nameSpec := hostSpec.Spec()
+
 	// insert state
-	rg.InsertState(sj, c.Domain, c.SOARname, c.Listener, c.Masters)
+	rg.InsertState(sj, c.Domain, c.SOARname, c.Listener, c.Masters, nameSpec)
```

Comment:
Actually, I had initially modeled it that way. I ended up doing it this way instead because it's consistent with the existing design. It also further highlights a weakness of the current design (I think we're passing in too many parameters here, especially those that will not change) that should be refactored when we re-do the storage implementation. So it should be exceedingly obvious that this is not the right approach and we need to change it.